src/sass/checkboxes/_base.scss

```diff
@@ -3,121 +3,112 @@
-  .#{$prefix}-checkbox-wrapper &:indeterminate:focus ~ .#{$prefix}-label::before {
+  input[type=checkbox]:focus + label::before,
+  input[type=checkbox]:focus ~ label::before {
     box-shadow:
-      0 0 0 .08rem $color-blue-600,
-      0 0 0 .1875rem $color-white-base,
-      0 0 0 .3125rem $color-blue-500;
+      0 0 0 ($spacing-xxs / 2) $color-black-500,
+      0 0 0 $spacing-xxs $color-white-base,
+      0 0 0 ($spacing-xxs * 1.5) $color-blue-600;
   }
 
-  /* stylelint-enable */
+  input[type=checkbox]:disabled {
+    cursor: default;
+  }
 
-  &:disabled + .#{$prefix}-label,
-  .#{$prefix}-checkbox-wrapper &:disabled ~ .#{$prefix}-label {
+  input[type=checkbox]:disabled + label,
+  input[type=checkbox]:disabled ~ label {
     color: $color-black-700;
     cursor: default;
   }
 
-  /* stylelint-disable */
-
-  /**
-   * Indeterminate state
-   */
+  input[type=checkbox]:disabled + label::before,
+  input[type=checkbox]:disabled ~ label::before {
+    background-color: $color-black-200;
+    box-shadow: 0 0 0 ($spacing-xxs / 2) $color-black-300;
+  }
 
-  &:indeterminate + .#{$prefix}-label::before,
-  .#{$prefix}-checkbox-wrapper &:indeterminate ~ .#{$prefix}-label::before {
-    background-image: url('data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgaGVpZ2h0PSIxNiIgdmlld0JveD0iMCAwIDE2IDE2Ij4KICAgIDxwYXRoIGZpbGw9IiNmZmZmZmYiIGQ9Ik0xNCw5SDJBMSwxLDAsMCwxLDIsN0gxNGExLDEsMCwwLDEsMCwyWiIvPgo8L3N2Zz4=');
+  input[type=checkbox]:indeterminate + label::before,
+  input[type=checkbox]:indeterminate ~ label::before {
     background-color: $color-blue-600;
-    box-shadow: 0 0 0 .08rem $color-blue-600;
+    background-image: url('data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHdpZHRoPSIxNiIgaGVpZ2h0PSIxNiIgdmlld0JveD0iMCAwIDE2IDE2Ij4KICA8cGF0aCBmaWxsPSIjZmZmZmZmIiBkPSJNMTMsOUgzQTEsMSwwLDAsMSwzLDdIMTNhMSwxLDAsMCwxLDAsMloiLz4KPC9zdmc+Cg==');
+    box-shadow: 0 0 0 ($spacing-xxs / 2) $color-blue-600;
   }
 
-  /* stylelint-enable */
-
-  &:indeterminate:disabled + .#{$prefix}-label::before,
-  .#{$prefix}-checkbox-wrapper &:indeterminate:disabled ~ .#{$prefix}-label::before {
+  input[type=checkbox]:indeterminate:disabled + label::before,
+  input[type=checkbox]:indeterminate:disabled ~ label::before {
     background-color: $color-black-200;
-    box-shadow: 0 0 0 .08rem $color-black-300;
+    box-shadow: 0 0 0 ($spacing-xxs / 2) $color-black-300;
+  }
+
+  input[type=checkbox]:checked:focus + label::before,
+  input[type=checkbox]:indeterminate:focus + label::before,
+  input[type=checkbox]:checked:focus ~ label::before,
+  input[type=checkbox]:indeterminate:focus ~ label::before {
+    box-shadow:
+      0 0 0 ($spacing-xxs / 2) $color-blue-600,
+      0 0 0 $spacing-xxs $color-white-base,
+      0 0 0 ($spacing-xxs * 1.5) $color-blue-600;
+  }
+
+  &__description {
+    color: $color-black-600;
```

Comment:
`$color-black-600` passes "normal text" contrast tests for AA but not AAA. If you use `$color-black-700` (`#3d4757`), then that passes both AA and AAA standards. But if you use 700, it's not much contrast against the base color of 800. So, I don't know what to recommend. Just wanted to outline the situation, in case that influences anything. I could be fine with whatever direction.

Reply:
As long as we're passing AA, I think we're good here. Of course AAA would be ideal, but given the visual contrast we're trying to achieve. I say leave it as is.